DSCResources/MSFT_xIPAddress/MSFT_xIPAddress.psm1

```diff
@@ -134,48 +130,162 @@ function ValidateProperties
 
     if(!([System.Net.Ipaddress]::TryParse($ip, [ref]0)))
     {
-       throw "IP Address *$IPAddress* is not in the correct format. Please correct the ipaddress in the configuration and try again"
+        throw ( @(
+            "IP Address *$IPAddress* is not in the correct format. Please correct the ipaddress "
+            "in the configuration and try again"
+            ) -join "" | Out-String
+        )
     }
 
     try
     {
         Write-Verbose -Message "Checking the IPAddress ..."
         #Get the current IP Address based on the parameters given.
-        $currentIP = Get-NetIPAddress -InterfaceAlias $InterfaceAlias -AddressFamily $AddressFamily -ErrorAction Stop
+        $currentIP = Get-NetIPAddress -InterfaceAlias $InterfaceAlias -AddressFamily `
+            $AddressFamily -ErrorAction Stop
+
+        # Build hash table of current settings
+        $currentSettings = @{}
```

Comment:
nit: Personally I prefer to see hashtables built like this:

````
        $currentSettings = @{
            IPAddress = $currentIP.IPAddress
            PrefixLength = $currentIP.PrefixLength
            InterfaceAlias = $currentIP.InterfaceAlias
        }
````

However, I'm being nitpicky so feel free to ignore me. :grinning: